core/unit_test/tools/include/ToolTestingUtilities.hpp

```diff
@@ -267,9 +267,19 @@ struct invoke_helper {
  */
 inline MatchDiagnostic check_match(event_vector::size_type events_scanned,
                                    const event_vector& events) {
-  return (events_scanned == events.size())
-             ? MatchDiagnostic{true}
-             : MatchDiagnostic{false, {"Wrong number of events encountered"}};
+// Workaround intel/17 internal error
+// See https://github.com/kokkos/kokkos/issues/4612
+#ifndef KOKKOS_COMPILER_INTEL
```

Comment:
```suggestion
#if defined(KOKKOS_COMPILER_INTEL) && (KOKKOS_COMPILER_INTEL < 1800)
```

Reply:
@dalg24 thanks for feedback, force pushed the update 54112f4fa8b1073f8a503f1ffa9e963dd8fc4711